docs/internals/variable_cleanup.rst

```diff
@@ -1,29 +1,31 @@
 .. index: variable cleanup
 
 *********************
-Cleaning Up Variables
+Zeroing of Padding
 *********************
 
-When a value is shorter than 256 bit, in some cases the remaining bits
-must be cleaned.
-The Solidity compiler is designed to clean such remaining bits before any operations
-that might be adversely affected by the potential garbage in the remaining bits.
-For example, before writing a value to  memory, the remaining bits need
-to be cleared because the memory contents can be used for computing
-hashes or sent as the data of a message call.  Similarly, before
-storing a value in the storage, the remaining bits need to be cleaned
-because otherwise the garbled value can be observed.
+When a value is shorter than 256 bit, in some cases the remaining (padding) bits
+must be zeroed. The Solidity compiler is designed to zero any padding bits before operations
+that might be adversely affected by garbage in the padding bits.
+For example, before writing a value to  memory, the padding bits need
+to be cleared because the memory contents could be used for computing
+hashes or sent as the data of a message call. Similarly, before
+storing a value in storage, padding bits need to be cleaned
+because otherwise their value can be observed, and may leak information.
```

Comment:
"Leak" sounds as if it was about security. I think it's really more about potentially affecting the results of operations. There's no memory protection so contract could get at these values anyway if it wanted to.

Reply:
Yes, the concern is about it being observed. But it's not a concern because the information is secret (as calling it a "leak" implies). It's because relying on it can make the behavior of your contract non-deterministic. While it's not really random, it may depend on low-level details of contract execution and compiler implementation details.

Reply:
For example the values before being cleaned could be different between the legacy and IR codegen. Or with and without the optimizer.